src/Group.cpp

```diff
@@ -0,0 +1,260 @@
+#include "global.h"
+#include "SongManager.h"
+#include "RageFile.h"
+#include "RageUtil.h"
+#include "RageFileManager.h"
+#include "RageLog.h"
+#include "Song.h"
+#include "SongCacheIndex.h"
+#include "SongUtil.h"
+#include "TitleSubstitution.h"
+#include "Group.h"
+
+#include <cstddef>
+#include <tuple>
+#include <vector>
+
+Group::Group() {
+    m_sDisplayTitle = "";
+    m_sSortTitle = "";
+    m_sPath = "";
+    m_sGroupName = "";
+    m_sTranslitTitle = "";
+    m_sSeries = "";
+    m_fSyncOffset = 0;
+    m_bHasPackIni = false;
+    m_iYearReleased = 0;
+    m_sBannerPath = "";
+    m_sCredits.clear();
+    m_sAuthorsNotes = "";
+}
+
+Group::~Group() {
+    SONGMAN->GetGroupGroupMap().clear();
+    m_sCredits.clear();
+}
+
+Group::Group(const RString& sDir, const RString& sGroupDirName) {
+    RString sPackIniPath = sDir + sGroupDirName + "/pack.ini";
+    RString credits = "";
```

Comment:
A lot of these fields need formatting fixes e.g. prefix with `s` + capitalize the first letter.